src/planner/planner.cpp

```diff
@@ -25,18 +34,56 @@ void Planner::PlanQuery(const BoundStatement &statement) {
   }
 }
 
-auto Planner::PlanExpression(const BoundExpression &expr, const AbstractPlanNode &child)
+auto Planner::PlanExpression(const BoundExpression &expr, const std::vector<const AbstractPlanNode *> &children)
     -> std::unique_ptr<AbstractExpression> {
   switch (expr.type_) {
     case ExpressionType::AGG_CALL:
       throw Exception("agg call should be handled by PlanSelect");
     case ExpressionType::COLUMN_REF: {
-      auto column_ref_expr = dynamic_cast<const BoundColumnRef &>(expr);
-      auto schema = child.OutputSchema();
-      // TODO(chi): differentiate multiple tables
-      uint32_t col_idx = schema->GetColIdx(column_ref_expr.col_);
-      auto col_type = schema->GetColumn(col_idx).GetType();
-      return std::make_unique<ColumnValueExpression>(0, col_idx, col_type);
+      const auto &column_ref_expr = dynamic_cast<const BoundColumnRef &>(expr);
```

Comment:
Why does `COLUMN_REF` expression have children? IMO, it should be the leaf node.

Reply:
If an expression is for join condition, e.g.

SELECT * from test_1 inner join test_2 on test_1.colA = test_2.col2

The plan will be like:

NestedLoopJoin condition = ColumnRef 0.0 = ColumnRef 1.2
  SeqScan colA, colB
  SeqScan col1, col2

where child_id is, when executor is using the expression, it picks from its nth children's mth column to get the data.

Reply:
I think this explanation is helpful, can you include a comment for context about what the `size == 1` and `size == 2` branches are assuming? (and fix the 1.2 -> 1.1) It can be a very short description, it's just that students might get scared off by trying to read the C++ directly without comments to break it up a bit

Reply:
updated :)